babel_edtf/__init__.py

```diff
@@ -174,6 +175,23 @@ def _format_edtf0_interval_naive(edtf_interval, format, locale):
         dt_start, dt_end, skeleton, fuzzy=True, locale=locale)
 
 
+def _validate_leap_year(edtf_date):
+    """Validate that dates on the 29th of February are on leap years."""
+    if isinstance(edtf_date, Date):
+        if (
+            edtf_date.precision == PRECISION_DAY
+            and edtf_date.day == '29'
+            and edtf_date.month == '02'
+            and not calendar.isleap(int(edtf_date.year))
```

Comment:
Remark:

Another way to validate the 3 lines above could be to rely on python-edtf's `days_in_month` like so:

```python
and int(edtf_date.day) > days_in_month(int(edtf_date.year), int(edtf_date.month))
```

Yet another option could be to try to convert parsed EDTF date to datetime, but this is less explicit and is executing more unnecessary code:

```python
try:
    struct_time_to_datetime(edtf_date.lower_strict())
    struct_time_to_datetime(edtf_date.upper_strict())
except ValueError:
    raise EDTFParseException('...')
```

Reply:
I like the current version, which reads very explicitly/naturally, so not much math to do in one's head :)